### PR TITLE
Add additional requirements for Reporting Engine

### LINKED ISSUE #54: Update Pentestportal Dependencies in preparation for RE 2.0.4 Deployment
# 💡 Summary #

RE 2.0.4 has been released and new system requirements have been pushed out. We'd like to request that the COOL pentestportal instances be updated to incorporate the various dependencies necessary for RE 2.0.4 to run properly on the system.

## Motivation and context ##

RE 2.0.4 is necessary for RVA engagements to be able to perform their day to day reporting procedures. As such, the tool must be able to run on the pentestportal instances deployed in the COOL operational environments. 

## Implementation notes ##

If an existing version of Node/NPM exists on the VM or if there are any leftover artifacts from previous installations of Node/NPM, the following commands should be run to purge them:
 
sudo rm -rf /usr/local/bin/npm /usr/local/share/man/man1/node* ~/.npm
sudo rm -rf /usr/local/lib/node*
sudo rm -rf /usr/local/bin/node*
sudo rm -rf /usr/local/include/node*
sudo apt-get purge nodejs npm
sudo apt autoremove
 
Installing Node v18.13.x and NPM v9.2.x:
 
curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
NODE_MAJOR=18
echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | sudo tee /etc/apt/sources.list.d/nodesource.list
apt-get update && apt-get install nodejs -y && apt-get install npm -y
Verify Node version: node --version
Verify NPM version: npm --version
 
Installing Python3-Django v2.2.x:
 
apt install python3-django
Verify python3-django version: python3 -m django --version
 
Additional requirements that currently already exist on the pentestportal instance include: Docker v20.10.12+, Docker Compose v2.2.3+, and Python v3.8.0+.
 
Verify Docker version: docker version
Verify Docker Compose version: docker compose version
Verify Python version: python3 --version

## Acceptance criteria ##

How do we know when this work is done?

- [x] Node v18.13.x is installed (actually 18.18.2)
- [x] NPM v9.2.x is installed (actually 9.8.1)
- [x] Python3-Django v2.2.x is installed (actually 2.2.28)
- [x] Docker v20.10.12+ is installed (actually 24.0.7)
- [x] Docker Compose v2.2.3+ is installed (actually 2.21.0)
- [x] Python v.3.8.0+ is installed (actually 3.9.2)

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request installs the following additional requirements for RE:
- Node.JS 18 (from the official package repo)
- The `python3-django` system package

## 💭 Motivation and context ##

Resolves #54.

## 🧪 Testing ##

All automated tests pass.  I also built a staging AMI with these changes and verified that it satisfies the acceptance criteria of #54.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

**Comment by dv4harr10** [comment]
Hi Team, Please confirm that the AWS Access Key ID Value credential is not hardcoded and accessible on the Readme.md file at line 93. Alternative suggestion can be to read the value from an environment variable or keep it in a separate, private file.

File: src/reporting_engine.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        - name: Update the cache with the Node goodness
+          ansible.builtin.package:
+            update_cache: yes
+            # This cache update can cause idempotence to fail, so tell
+            # molecule to ignore any changes this task produces when
+            # testing idempotence.
+          tags:
+            - molecule-idempotence-notest
+
+        - name: Install Node.JS
```
Comment (dav3r) [inline src/reporting_engine.yml]: This appears to be the [preferred nomenclature](https://nodejs.org/en/):
```suggestion
        - name: Install Node.js
```

**Review by dav3r — APPROVED** [review]
All good, aside from a minor quibble

**Comment by jsf9k** [comment]
> Hi Team, Please confirm that the AWS Access Key ID Value credential is not hardcoded and accessible on the Readme.md file at line 93. Alternative suggestion can be to read the value from an environment variable or keep it in a separate, private file.

Fear not, @dv4harr10!  Those are dummy values.

**Comment by dv4harr10** [comment]
>> Hi Team, Please confirm that the AWS Access Key ID Value credential is not hardcoded and accessible on the Readme.md file at line 93. Alternative suggestion can be to read the value from an environment variable or keep it in a separate, private file.

> Fear not, @dv4harr10! Those are dummy values.

Got it, Thanks

**Review by dv4harr10 — APPROVED** [review]

**Pull request merged**